bazel/envoy_build_system.bzl

```diff
@@ -213,6 +214,11 @@ def envoy_cmake_external(
     # On Windows, we don't want to explicitly set CMAKE_BUILD_TYPE,
     # rules_foreign_cc will figure it out for us
     cache_entries_no_build_type = {key: cache_entries[key] for key in cache_entries.keys() if key != "CMAKE_BUILD_TYPE"}
+    cache_entries_no_build_type_debug = dict(cache_entries_no_build_type)
```

Comment:
Suggest simplifying this by just having a function to filter out Windows entries, i.e. `filter_windows_keys` and apply it directly in the `cache_entries` dict literal below. That way, we're not tracking 4 different dictionaries, which is a bit confusing.